src/main/resources/com/google/api/codegen/go/doc.snip

```diff
@@ -21,11 +21,13 @@
         "google.golang.org/grpc/metadata"
     )
 
-    func insertXGoog(ctx context.Context, val []string) context.Context {
-        md, _ := metadata.FromOutgoingContext(ctx)
-        md = md.Copy()
-        md["x-goog-api-client"] = val
-        return metadata.NewOutgoingContext(ctx, md)
+    func insertMetadata(ctx context.Context, md map[string][]string) context.Context {
+        out, _ := metadata.FromOutgoingContext(ctx)
+        out = out.Copy()
+        for k, v := range md {
+            out[k] = v
+        }
```

Comment:
`metadata.Join(out, md)`? No need to `Copy`, `Join` always allocates a new one.

Reply:
Good point, done